.github/workflows/build-test.yml

```diff
@@ -19,43 +19,108 @@ jobs:
 
-  tests-pekko-http:
-    name: Sbt Tests Pekko HTTP
-#    needs:
-#      - "check-code-style"
-    uses: playframework/.github/.github/workflows/cmd.yml@v3
-    with:
-      java: 17, 11
-      scala: 2.13.x, 3.x
-      cmd: |
-        # sudo add-apt-repository -y ppa:ondrej/php && sudo apt-get -qq update && sudo apt-get install -y libsodium-dev # for secure session examples
-        ./test.sh
-
-  tests-netty-http:
-    name: Sbt Tests Netty HTTP
-#    needs:
-#      - "check-code-style"
-    uses: playframework/.github/.github/workflows/cmd.yml@v3
-    with:
-      java: 17, 11
-      scala: 2.13.x, 3.x
-      cmd: |
-        ./test.sh --netty
+  changes:
+    if: github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      examples: ${{ steps.filter.outputs.changes }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            java-chatroom: play-java-chatroom-example/**
+            java-compile-di: play-java-compile-di-example/**
+            java-dagger2: play-java-dagger2-example/**
+            java-ebean: play-java-ebean-example/**
+            java-fileupload: play-java-fileupload-example/**
+            java-forms: play-java-forms-example/**
+            java-grpc: play-java-grpc-example/**
+            java-hello-world: play-java-hello-world-tutorial/**
+            java-jpa: play-java-jpa-example/**
+            java-pekko-cluster: play-java-pekko-cluster-example/**
+            java-rest-api: play-java-rest-api-example/**
+            java-starter: play-java-starter-example/**
+            java-streaming: play-java-streaming-example/**
+            java-websocket: play-java-websocket-example/**
+            scala-anorm: play-scala-anorm-example/**
+            scala-chatroom: play-scala-chatroom-example/**
+            scala-compile-di: play-scala-compile-di-example/**
+            scala-fileupload: play-scala-fileupload-example/**
+            scala-forms: play-scala-forms-example/**
+            scala-grpc: play-scala-grpc-example/**
+            scala-hello-world: play-scala-hello-world-tutorial/**
+            scala-isolated-slick: play-scala-isolated-slick-example/**
+            scala-log4j2: play-scala-log4j2-example/**
+            scala-macwire-di: play-scala-macwire-di-example/**
+            scala-rest-api: play-scala-rest-api-example/**
+            scala-secure-session: play-scala-secure-session-example/**
+            scala-slick: play-scala-slick-example/**
+            scala-starter: play-scala-starter-example/**
+            scala-streaming: play-scala-streaming-example/**
+            scala-tls: play-scala-tls-example/**
+            scala-websocket: play-scala-websocket-example/**
 
-  gradle-tests:
-    name: Gradle Tests
-    uses: playframework/.github/.github/workflows/cmd.yml@v3
+  java-chatroom:
```

Comment:
So does this mean you have to duplicate this section for each samples, like 31 times? Each time basically the same except a different sample name?

Reply:
Yes, you are right. Potentially, we can try to pass `needs.changes.outputs.examples` array as another dimension for matrix job, but because I need to do that only once, currently I prefer duplicate jobs per example to have a full control of build logic for every example.

Reply:
It's a reasonable cost to me to get the target. But of course it's discussable 😉

Reply:
Thinking...